homeassistant/components/zwave_js/discovery.py

```diff
@@ -186,36 +188,30 @@ class ZWaveDiscoverySchema:
         ),
     ),
     # climate
+    # thermostats supporting mode (and optional setpoint)
     ZWaveDiscoverySchema(
         platform="climate",
-        device_class_generic={"Thermostat"},
-        device_class_specific={
-            "Setback Thermostat",
-            "Thermostat General",
-            "Thermostat General V2",
-            "General Thermostat",
-            "General Thermostat V2",
-        },
         primary_value=ZWaveValueDiscoverySchema(
             command_class={CommandClass.THERMOSTAT_MODE},
             property={"mode"},
             type={"number"},
         ),
     ),
-    # climate
-    # setpoint thermostats
+    # thermostats supporting setpoint only (and thus not mode)
     ZWaveDiscoverySchema(
         platform="climate",
-        device_class_generic={"Thermostat"},
-        device_class_specific={
-            "Setpoint Thermostat",
-            "Unused",
-        },
         primary_value=ZWaveValueDiscoverySchema(
             command_class={CommandClass.THERMOSTAT_SETPOINT},
             property={"setpoint"},
             type={"number"},
         ),
+        absent_values=[  # mode may not be present to prevent dupes
```

Comment:
Does `may` mean `must`?
```suggestion
        absent_values=[  # mode must not be present to prevent dupes
```

Reply:
correct, good catch